cnd/src/http_api/routes/index.rs

```diff
@@ -66,28 +65,26 @@ pub async fn post_herc20_halight_bitcoin(
         .map_err(problem::from_anyhow)
         .map_err(warp::reject::custom)?;
 
-    let swap_params: Herc20HalightBitcoinCreateSwapParams = body.clone().into();
-
     let swap_id = LocalSwapId::default();
     let reply = warp::reply::reply();
 
-    let swap = CreatedSwap::<herc20::CreatedSwap, halight::CreatedSwap> {
-        swap_id,
-        alpha: body.alpha.into(),
-        beta: body.beta.into(),
-        peer: body.peer.into(),
-        address_hint: None,
-        role: body.role.0,
-    };
-
+    let swap = body.to_created_swap(swap_id);
```

Comment:
`ToCreatedSwap` trait is defined in this file. I understand it allows the conversion to a `db::CreatedSwap<A,B>` struct that can be saved in the db.

At first read, it is not clear why a `Body` is converted to a `CreatedSwap` using a local trait. Maybe keeping the `db` module in the usage path would help clarity: `db::CreatedSwap`.

Also, I am not sure the trait is necessary here, to me it just adds indirection. Instead, you can simply add a `fn to_create_swap` to `impl Block<Herc..,Halight..>` because `Body<A,B>` is defined in the same file (and hence you can have a `impl` block for it here).

Reply:
Implemented already, GitHub fail by me, Thomas and I had this conversation and I marked it as resolved before force pushing. Sorry for wasting your time.